### PR TITLE
Stack position

### PR DESCRIPTION
**Pull request by kevin-logan**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -137,16 +137,16 @@ end
 You can call `example_callable_from_cpp` from C++ like so:
 ```C++
 glua.RunFile("example.lua");
-glua.CallScriptFunction("example_callable_from_cpp", 1337, "herpaderp");
+auto retvals = glua.CallScriptFunction("example_callable_from_cpp", 1337, "herpaderp");
 ```
-`GluaBase::CallScriptFunction` takes first the name of the function to call, and then any arguments you wish to pass to that function. The return value, however, is pushed onto the stack and not automatically returned for you (as a lua function could actually returned multiple arguments, which is precisely what is happening in this case).
+`GluaBase::CallScriptFunction` takes first the name of the function to call, and then any arguments you wish to pass to that function. The return valuesare pushed onto the stack and automatically returned for you (as a lua function could actually returned multiple arguments, which is precisely what is happening in this case).
```
Comment (PrestonFrom) [inline README.md]: 'valuesare' should be 'values are'

File: inc/glua/StackPosition.h [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+     * @throws std::bad_optional_access if this StackPosition no longer
+     * holds an index (either it's been moved or Release was called).
+     *
+     * @throws std::out_of_range if there is no child with the given
+     * index
+     */
+    auto SafePushChild(size_t child_index) -> StackPosition;
+    /**
+     * Pushes the child of this object with the given key onto the stack
+     * and returns a StackPosition referring to it.If the child doesn't
```
Comment (PrestonFrom) [inline inc/glua/StackPosition.h]: "it.If" -> "it. If"

File: inc/glua/StackPosition.h [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    /**
+     * @tparam Type the type to convert this item to
+     * @return The value of the item this position refers to converted to
+     * the requested type
+     */
+    template<typename Type>
+    auto As() -> Type;
+
+    /**
+     * @tparam Type the Type this value against
```
Comment (PrestonFrom) [inline inc/glua/StackPosition.h]: the Type this value is checked against?

File: README.md (hunk 1 above)
Comment (kevin-logan) [reply README.md]: done

File: inc/glua/StackPosition.h (hunk 2 above)
Comment (kevin-logan) [reply inc/glua/StackPosition.h]: done

File: inc/glua/StackPosition.h (hunk 3 above)
Comment (kevin-logan) [reply inc/glua/StackPosition.h]: done

**Review by PrestonFrom — APPROVED** [review]

**Pull request merged**